core/js/js.js

```diff
@@ -770,6 +770,15 @@ var OCP = {},
 		return false;
 	},
 
+	/**
+	 * Returns the user's locale as a BCP 47 compliant language tag
+	 *
+	 * @return {String} locale string
+	 */
+	getCanonicalLocale: function() {
+		return this.getLocale().replace(/_/g, '-');
```

Comment:
getLocale() can be undefined
```suggestion
		var locale = this.getLocale();
		return typeof locale === 'string' ? locale.replace(/_/g, '-') : locale;
```